network/firewall.py

```diff
@@ -0,0 +1,396 @@
+import logging
+import json
+from typing import List, Dict
+
+import network.network as nw
+from nftables import Nftables
+
+logger = logging.getLogger(__name__)
+
+nft = Nftables()
+nft.set_json_output(True)
+nft.set_stateless_output(True)
+nft.set_service_output(False)
+nft.set_reversedns_output(False)
+nft.set_numeric_proto_output(True)
+
+
+class Firewall:
+    firewall_has_been_setup = False
+    postrouting_info: Dict[str, str] = {}
+    forwarding_info: Dict[str, str] = {}
+    vm_info: Dict[int, Dict] = {}
+
+    @classmethod
+    def execute_json_nft_commands(cls, commands: List[Dict]) -> int:
+        """Executes a list of nftables commands, and returns the exit status"""
+        commands_dict = {"nftables": commands}
+        try:
+            logger.debug("Validating nftables rules")
+            nft.json_validate(commands_dict)
+        except Exception as e:
+            logger.error(f"Failed to verify nftables rules: {e}")
+
+        logger.debug("Inserting nftables rules")
+        rc, output, error = nft.json_cmd(commands_dict)
```

Comment:
What does `rc` stand for ? Should the answert to that question be in the code for future readings ?

Reply:
rc = return code. For me that was very obvious when I saw that exact code in example code on how to use the nftables lib: https://github.com/aborrero/python-nftables-tutorial/blob/main/nft-read-json.py

Together with the very next line where it tests for non-zeroness and if not outputs the error, I don't believe changing it would make it really clearer. naming it return_code or similar only would make it more verbose in my opinion, not clearer.